livekit-agents/livekit/agents/codecs/mp3.py

```diff
@@ -53,31 +62,42 @@ async def _run(self):
                 self._output_queue.put_nowait(None)
                 break
 
-            result = await asyncio.to_thread(self._decode_input, input)
-            # If error decoding, skip it
-            if result is None:
-                continue
-            self._output_queue.put_nowait(result)
+            result_frames = await asyncio.to_thread(self._decode_input, input)
+            for frame in result_frames:
+                await self._output_queue.put(frame)
 
     def _decode_input(self, input: bytes):
         packets = self._codec.parse(input)
-        result = b""
+        result_frames: List[rtc.AudioFrame] = []
         for packet in packets:
             try:
                 decoded = self._codec.decode(packet)
                 for frame in decoded:
+                    nchannels = len(frame.layout.channels)
```

Comment:
This isn't valid, you're stil using the frames with nchannels > 1
I think making the API sync will makes everything much simpler. Error handling is starting to be hard here no?

Reply:
fair, i'll just make it all sync in a fast-follow PR